### ISSUE TITLE
How to decode encoded class label from webdataset?

### ISSUE THREAD

**Issue by LuoXin-s** [NVIDIA/DALI#4839]
### Describe the question.

I write `class_idx` to webdataset with `"cls": str(class_idx).encode("ascii")`.

After I read image and label from `fn.readers.webdataset` through `images, labels = fn.readers.webdataset()`.

I'm wondering how could I decode label properly, otherwise, I‘ll get a Runtime Error:

`Assert on "IsDenseTensor()" failed: The batch must be representable tensor - it must has uniform shape and be allocated in contiguous memory.`

I make sure there is no problem regarding images.

### Check for duplicates

- [X] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

**Comment by LuoXin-s** [comment]
This is my index file. Every label would occupy 3 bytes.
<img width="610" alt="image" src="https://user-images.githubusercontent.com/20135317/237037410-3c9f4dd0-101f-4284-be44-1d9a98025a26.png">

**Comment by LuoXin-s** [comment]
I guess the encode method used in generate webdataset may not compatible with DALI, though it is default encode method for class label in webdataset, see [here](https://github.com/webdataset/webdataset/blob/039d74319ae55e5696dcef89829be9671802cf70/webdataset/writer.py#L148).

To solve this problem, encode class label as Binary encoding in 2 bytes, then load it with dtypes=types.INT16 would be a graceful way?

**Comment by LuoXin-s** [comment]
I would try using struct.pack("i", class_idx) to encode my label. Then, there is a subtle problem, struct.pack use little-endian in default, fn.readers.webdataset will interpret bytes in little-endian order also?

**Comment by LuoXin-s** [comment]
Hi @jantonguirao , how can I apply `struct.unpack` to these 4 bytes. What's more I want transform them into integer in pipeline so I do not have to modify my training code.

**Comment by jantonguirao** [comment]
Can you share more details about your pipeline? What operations would you perform on that integer?

**Comment by LuoXin-s** [comment]
yeah, actually I just want to read this label integer from a webdataset file.

For example, I use following Iterator in pytorch:
```python
data_loader = DALIGenericIterator(pipe, ["img", "label"], reader_name="Reader", last_batch_policy=LastBatchPolicy.DROP, auto_reset=True)
```
I just want data loaded from data_loader is behaved like a normal dataloader. That is, data['label'] is a batch of integer other than a batch of 4 bytes tensors. I would like the transformation is taken in pipe.

**Comment by LuoXin-s** [comment]
I see python operators such as `TorchPythonFunction` should solve my problem, but they would harm performance of pipeline.

**Comment by LuoXin-s** [comment]
😂Ok, I found `fn.readers.webdataset` interpret bytes with little-endian order implicitly, so `fn.readers.webdataset(dtypes=[types.UINT8, types.INT32]) can satisfy my request.`

**Comment by renmin56** [comment]
> 😂Ok, I found `fn.readers.webdataset` interpret bytes with little-endian order implicitly, so `fn.readers.webdataset(dtypes=[types.UINT8, types.INT32]) can satisfy my request.`

how to write class_idx to webdataset ?   "cls": str(class_idx).encode("ascii")   or  struct.pack("i", class_idx) ?

**Comment by renmin56** [comment]
> 😂Ok, I found `fn.readers.webdataset` interpret bytes with little-endian order implicitly, so `fn.readers.webdataset(dtypes=[types.UINT8, types.INT32]) can satisfy my request.`

how to write class_idx to webdataset ? "cls": str(class_idx).encode("ascii") or struct.pack("i", class_idx) ?  thanks

**Comment by renmin56** [comment]
> 😂Ok, I found `fn.readers.webdataset` interpret bytes with little-endian order implicitly, so `fn.readers.webdataset(dtypes=[types.UINT8, types.INT32]) can satisfy my request.`

have the same problem, thanks
how to write class_idx to webdataset ? "cls": str(class_idx).encode("ascii") or struct.pack("i", class_idx) ?
